### ISSUE TITLE
Support autoboot for r4isdhc.com.cn flashcard

### ISSUE THREAD

**Issue by glatavento** [DS-Homebrew/TWiLightMenu#1380]
Or is there any way to boot TWiLightMenu from a CN card automatically? [Here](https://drive.google.com/file/d/1lwfKXFs4NVDYr4eyGkGMPWMOx8BJIK_o/view?usp=sharing) are some firmwares for CN card.

**Comment by hansoljin** [comment]
It is questionable whether this will even be possible. This specific cart has a header check, where if a certain portion of the ROM does not match specifically what the cartridge wants, it refuses to boot. This means the header of the kernel's main autoboot file needs to be extracted and applied to `BOOT.NDS` for this autoboot to work. That's too much to deal with. Add to that, the autoboot file must be encrypted (though the encryption key is public.)

A suggestion is to create a payload that matches these specifications, but would redirect to another ROM directly. This would be a great idea, but is quite a gamble. Many flashcards have protection built for this sort of thing (why they do that is beyond me) but that would certainly make things harder.

Another suggestion is to rebuild Taiju Yamada's Wood EX4 to autoboot TWiLight Menu++ instead. This would add a few seconds to the boot time, as one would boot the usual Wood kernel and have it directly load TWiLight. This is way too hacky a workaround to actually include this in the release.

The bottom line is that it would be too much to work around just to get TWiLight to boot automatically. I'm not entirely sure whether that is feasible for this project.

**Comment by lifehackerhansol** [comment]
I have created an autoboot payload for this cart on my repo here: https://github.com/lifehackerhansol/flashcard-bootstrap

This automatically DLDI patches `BOOT.NDS` and loads it, and would be a much more effective solution.

**Comment by lifehackerhansol** [comment]
Fixed with #1493.

**Comment by emilek133** [comment]
> The autoboot is already implemented in the latest TWiLight Menu++. My comment above is no longer required.
On page wiki ds- homebrew is a tutorial on how to install TWL ++ on a flash card. I follow the steps:

__Autobooting TWiLight Menu++ #
Extract the content(s) of Autoboot/(your flashcard) to the root of the flashcard’s microSD card
Skip this, if you don’t see your flashcard__

I checked all the matching _DSMENU files from the R4 directories.
Unfortunately, none starts TWL ++ automatically.

**Comment by emilek133** [comment]
Thank you for your help. Works great on the new twl ++ version
